tests/test_bpa_register.py

```diff
@@ -0,0 +1,220 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from auth.config import Settings
+from tests.datagen import AccessTokenPayloadFactory
+
+VALID_REGISTRATION_DATA = {
```

Comment:
Once you have a schema for the registration data, generate it like we do with the other data in `datagen` (you can still specify whichever fields you actually need for individual tests)